app/models/manageiq/providers/embedded_ansible/provider.rb

```diff
@@ -6,7 +6,7 @@ class ManageIQ::Providers::EmbeddedAnsible::Provider < ::Provider
   has_one :automation_manager,
           :foreign_key => "provider_id",
           :class_name  => "ManageIQ::Providers::EmbeddedAnsible::AutomationManager",
-          :dependent   => :destroy,
+          :dependent   => :destroy,   # to be removed after ansible_tower side code is updated
```

Comment:
Why?

Reply:
That's now being explicitly called in provider.destroy